src/main/java/edu/ie3/util/StringUtils.java

```diff
@@ -102,4 +114,42 @@ public static String quote(String input) {
   public static String cleanString(String input) {
     return input.replaceAll("[^\\w]", "_");
   }
+
+  /**
+   * Quotes a given string that contains special characters to comply with the csv specification RFC
+   * 4180 (https://tools.ietf.org/html/rfc4180). Double quotes in JSON strings are escaped with the
+   * same character to make the csv data readable later.
+   *
+   * @param inputString string that should be converted to a valid rfc 4180 string
+   * @param csvSep separator of the csv file
+   * @return a csv string that is valid according to rfc 4180
+   */
+  public static String csvString(String inputString, String csvSep) {
+    if (needsCsvRFC4180Quote(inputString, csvSep)) {
+      /* clean the string by first quoting start and end of the string and then replace all double quotes
+       * that are followed by one or more double quotes with single double quotes */
+      String quotedStartEndString = quote(inputString).replaceAll("\"\"*", "\"");
+      /* get everything in between the start and end quotes and replace single quotes with double quotes */
+      String stringWOStartEndQuotes =
+          quotedStartEndString
+              .substring(1, quotedStartEndString.length() - 1)
+              .replaceAll("\"", "\"\"");
+      /* finally add quotes to the strings start and end again */
+      return quote(stringWOStartEndQuotes);
+    } else return inputString;
+  }
+
+  /**
+   * Check if the provided string needs to be quoted according to the csv specification RFC 4180
+   *
+   * @param inputString the string that should be checked
+   * @param csvSep separator of the csv file
+   * @return true of the string needs to be quoted, false otherwise
+   */
+  private static boolean needsCsvRFC4180Quote(String inputString, String csvSep) {
+    return inputString.contains(csvSep)
+        || inputString.contains(",")
```

Comment:
From my point of view, this condition is not necessary, as a "," has not to be escaped, when `csvSep != ","`.

Reply:
Disagree at this point. 

> Fields containing line breaks (CRLF), double quotes, and commas
       should be enclosed in double-quotes

according to [RFC 4180, 2.6](https://tools.ietf.org/html/rfc4180)

Reply:
😯